src/Model/Notify/Type.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Friendica\Model\Notify;
+
+/**
+ * Enum for different types of the Notify
+ */
+class Type
+{
+	/** @var int Notification about a introduction */
+	const INTRO = 1;
+	/** @var int Notification about a confirmed introduction */
+	const CONFIRM = 2;
+	/** @var int Notification about a post on your wall */
+	const WALL = 4;
+	/** @var int Notification about a followup comment */
+	const COMMENT = 8;
+	/** @var int Notification about a private message */
+	const MAIL = 16;
+	/** @var int Notification about a friend suggestion */
+	const SUGGEST = 32;
+	/** @var int Unknown Notification
+	 * @deprecated
+	 */
+	const PROFILE = 64;
+	/** @var int Notification about being tagged in a post */
+	const TAG_SELF = 128;
+	/** @var int Notification about being tagged
+	 * @deprecated
+	 */
+	const TAG_SHARE = 256;
+	/** @var int Notification about getting poked/prodded/etc. */
+	const POKE = 512;
+	/** @var int Notification about either a contact had posted something directly or the contact is a mentioned forum */
+	const SHARE = 1024;
+
+	/** @var int Global System notifications */
+	const SYSTEM = 32768;
+
+	/**
+	 * Check if the notification type matches the needle
+	 *
+	 * @param int $haystack The base haystack
+	 * @param int $needle   The needle to search
+	 *
+	 * @return bool True, if the notification needle matches the haystack
+	 */
+	public static function match(int $haystack, int $needle)
+	{
+		return $haystack & $needle;
```

Comment:
I see, I know it from the JAVA world where we would have it like
`$notify->getType()->match(Type::INTRO)` for example
I thought too much in this direction ^^

Reply:
Java has a specific all-out object-method approach that explains this, the same way all strings are objects and methods are performed on them.

PHP opts for convenience with low-level language constructs to avoid this rather verbose style.

Reply:
reverted whole commit, nothing left from this construct ;)

Reply:
Yeah, I know from dev-team members that they create elegant constructs to avoid any kind of improper usage of types, enums, atomar types etc. .. And one thing, we really, really miss in PHP/our codebase:
`List<T>`, which would completely would be able to replace our `Collection` construct in a more global way (and things like `class Repository returns <T> {}` which would implicit type-safe the return types instead of our methods, we have to overwrite each time ^^.. Sadly that there are not equal design patterns to fulfill the same purpose in PHP :-)